src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,66 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunStartupBenchmark(string[] args, ProcessStartInfo psi, PreDel pre, MidDel mid, PostDel post)
+        {
+            pre(_performanceTestConfig, psi);
```

Comment:
Spoke with @jorive offline; I was convinced otherwise. :smile: The user is free to do one-time setup in their own C# code that is the harness; therefore we have no reason to support anything other than per-iteration setup. Ergo, I vote that we should move the pre (and post below) to happen per iteration.